CRM/Report/Form/Event/Income.php

```diff
@@ -111,18 +111,27 @@ public function buildEventReport($eventIDs) {
 
     $groupBy = CRM_Contact_BAO_Query::getGroupByFromSelectColumns($select, 'civicrm_event.id');
     $sql = "
-            SELECT  " . implode(', ', $select) . ",
-                    SUM(civicrm_participant.fee_amount) as total,
-                    COUNT(civicrm_participant.id)       as participant
-
-            FROM       civicrm_event
-            LEFT JOIN  civicrm_option_value
-                   ON  ( civicrm_event.event_type_id = civicrm_option_value.value AND
-                         civicrm_option_value.option_group_id = {$optionGroupId} )
-            LEFT JOIN  civicrm_participant ON ( civicrm_event.id = civicrm_participant.event_id
-                       {$activeParticipantClause} AND civicrm_participant.is_test  = 0 )
-
-            WHERE      civicrm_event.id IN( {$eventID}) {$groupBy}";
+      SELECT " . implode(', ', $select) . ",
+        SUM(civicrm_participant.fee_amount) as total,
+        COUNT(civicrm_participant.id) as participant
+      FROM civicrm_event
+      LEFT JOIN civicrm_option_value
+        ON civicrm_event.event_type_id = civicrm_option_value.value 
+        AND civicrm_option_value.option_group_id = {$optionGroupId}
+      LEFT JOIN (
+        SELECT civicrm_participant.*
+        FROM civicrm_participant
+        JOIN civicrm_contact
+          ON civicrm_participant.contact_id = civicrm_contact.id
+          AND civicrm_contact.is_deleted = 0
+        WHERE civicrm_participant.event_id IN ({$eventID})
```

Comment:
Since you are using INNER JOIN on single table can we directly add where clause to Join clause?

```
JOIN civicrm_contact
          ON civicrm_participant.contact_id = civicrm_contact.id
          AND civicrm_contact.is_deleted = 0
        AND civicrm_participant.event_id IN ({$eventID})
```